warehouse/forklift/legacy.py

```diff
@@ -1142,7 +1142,9 @@ def file_upload(request):
                 "provided signature has been discarded."
             )
             send_gpg_signature_uploaded_email(
-                request, request.user, project_name=project.name
+                request,
+                request.user if request.user else project.users,
```

Comment:
NB: Passing in either a scalar `user` or `list[users]` here is allowed by `@email`-decorated APIs, like `send_gpg_signature_uploaded_email`.